src/components/autoComplete.js

```diff
@@ -381,6 +401,7 @@
         freeSolo={freeSolo}
         autoSelect={freeSolo}
         options={results}
+        filterOptions={filterOptions}
```

Comment:
Are we now fetching the entire collection in multiple data api calls? That's not entirely what I expected here. If you remove the code on line 393: 

`
          if (!freeSolo) {
            return;
          }
`

The component will already behave as expected. Loading entire collections can also result in degraded performance when using it on large collections.